src/commonMain/kotlin/exchange.dydx.abacus/processor/input/TransferInputProcessor.kt

```diff
@@ -0,0 +1,371 @@
+package exchange.dydx.abacus.processor.input
+
+import exchange.dydx.abacus.calculator.v2.TransferInputCalculatorV2
+import exchange.dydx.abacus.output.input.DepositInputOptions
+import exchange.dydx.abacus.output.input.InputType
+import exchange.dydx.abacus.output.input.SelectionOption
+import exchange.dydx.abacus.output.input.TransferInputResources
+import exchange.dydx.abacus.output.input.TransferInputSize
+import exchange.dydx.abacus.output.input.TransferOutInputOptions
+import exchange.dydx.abacus.output.input.TransferType
+import exchange.dydx.abacus.output.input.WithdrawalInputOptions
+import exchange.dydx.abacus.processor.router.IRouterProcessor
+import exchange.dydx.abacus.protocols.ParserProtocol
+import exchange.dydx.abacus.responses.ParsingError
+import exchange.dydx.abacus.state.changes.Changes
+import exchange.dydx.abacus.state.changes.StateChanges
+import exchange.dydx.abacus.state.internalstate.InternalInputState
+import exchange.dydx.abacus.state.internalstate.InternalTransferInputState
+import exchange.dydx.abacus.state.internalstate.InternalWalletState
+import exchange.dydx.abacus.state.internalstate.safeCreate
+import exchange.dydx.abacus.state.manager.ExchangeConfig.exchangeList
+import exchange.dydx.abacus.state.manager.V4Environment
+import exchange.dydx.abacus.state.model.TransferInputField
+import exchange.dydx.abacus.utils.IList
+import exchange.dydx.abacus.utils.IMutableList
+import kollections.iListOf
+import kollections.iMutableListOf
+import kollections.toIList
+import kollections.toIMap
+
+internal class TransferInputProcessor(
+    private val parser: ParserProtocol,
+    private val calculator: TransferInputCalculatorV2 = TransferInputCalculatorV2(parser = parser),
+    private val routerProcessor: IRouterProcessor,
+    private val environment: V4Environment?
+) {
+    fun transfer(
+        inputState: InternalInputState,
+        walletState: InternalWalletState,
+        data: String?,
+        inputField: TransferInputField,
+        subaccountNumber: Int = 0
+    ): InputProcessorResult {
+        val error: ParsingError? = null
+
+        if (inputState.currentType != InputType.TRANSFER) {
+            inputState.currentType = InputType.TRANSFER
+
+            inputState.transfer.size = null
+            inputState.transfer.fastSpeed = false
+            inputState.transfer.fee = null
+            inputState.transfer.exchange = null
+            inputState.transfer.chain = null
+            inputState.transfer.token = null
+            inputState.transfer.address = null
+            inputState.transfer.memo = null
+            inputState.transfer.depositOptions = null
+            inputState.transfer.withdrawalOptions = null
+            inputState.transfer.transferOutOptions = null
+            inputState.transfer.summary = null
+            inputState.transfer.resources = null
+            inputState.transfer.route = null
+
+            inputState.transfer.type = TransferType.deposit
```

Comment:
is this transferType just the `default` before it gets overwritten in the `when` block?

Reply:
Yes, it's overwritten only when the inputField is "TransferInputField.type".   It's the same logic as before.

Reply:
Sounds good! Didn't realize that we defaulted to a transferType instead of leaving as null